.travis.yml

```diff
@@ -28,122 +23,131 @@ jobs:
     - # same stage, parallel job
       os: linux
-      name: Ubuntu AMD64 Integration Test
+      name: Linux AMD64 Integration Test
+      arch: amd64
       script:
-        - ./scripts/travis/integration_test.sh
+        - mule build-linux-amd64
+        - scripts/travis/travis_wait.sh 120 mule build-linux-amd64-integration
     - # same stage, parallel job
-      name: External ARM64 Build
+      name: Linux ARM64 Unit Test
       os: linux
-      env:
-        - BUILD_TYPE: "external_build"
-        - TARGET_PLATFORM: "linux-arm64"
-      addons:
-        apt:
-          packages:
-            - awscli
+      arch: arm64
+      if: type = pull_request
       script:
-        - scripts/travis/external_build.sh ./scripts/travis/build_test.sh
+        - scripts/travis/travis_wait.sh 90 mule build-linux-arm64-shorttest
     - # same stage, parallel job
-      name: External ARM64 Integration Test
+      name: Linux ARM64 Unit Test
       os: linux
-      env:
-        - BUILD_TYPE: "external_build"
-        - TARGET_PLATFORM: "linux-arm64"
-      addons:
-        apt:
-          packages:
-            - awscli
+      arch: arm64
+      if: type != pull_request
       script:
-        - scripts/travis/external_build.sh ./scripts/travis/integration_test.sh
+        - scripts/travis/travis_wait.sh 120 mule build-linux-arm64-fulltest
     - # same stage, parallel job
-      os: osx
-      osx_image: xcode11
-      name: MacOS AMD64 Build
+      name: Linux ARM64 Integration Test
+      os: linux
+      arch: arm64
       script:
-        - scripts/travis/build_test.sh
+        - mule build-linux-arm64
+        - scripts/travis/travis_wait.sh 120 mule build-linux-arm64-integration
     - # same stage, parallel job
       os: osx
       osx_image: xcode11
-      name: MacOS AMD64 Integration Test
-      script:
-        - ./scripts/travis/integration_test.sh
-
-    - stage: build_release
-      os: linux
-      name: Ubuntu AMD64 Build
-      script:
-        - ./scripts/travis/build_test.sh
-    - # same stage, parallel job
-      os: linux
-      name: Ubuntu AMD64 Integration Test
+      arch: amd64
+      name: MacOS AMD64 Unit Test
+      go:
+        - "1.12"
+      go_import_path: github.com/algorand/go-algorand
+      language: go
+      before_install:
+        - brew update
+        - brew upgrade pyenv
+        - pyenv install 3.7.6
+        - pyenv local  3.7.6
+      if: type = pull_request
       script:
-        - ./scripts/travis/integration_test.sh
+        - scripts/travis/travis_wait.sh 90 mule build-local-shorttest
     - # same stage, parallel job
       os: osx
       osx_image: xcode11
-      name: MacOS AMD64 Build
+      arch: amd64
+      name: MacOS AMD64 Unit Test
+      go:
+        - "1.12"
+      go_import_path: github.com/algorand/go-algorand
+      language: go
+      before_install:
+        - brew update
+        - brew upgrade pyenv
+        - pyenv install 3.7.6
+        - pyenv local  3.7.6
```

Comment:
One trick we used to avoid specifying all of this for different architectures in travis, was having a portable `configure_dev.sh` script. If you wrap `mule build-local-shorttest` in a script. If you add a `configure_mule.sh` script, and wrap the mule calls in a script which calls `configure_mule.sh` you can reduce some of this duplicate setup code.

Reply:
Oh that could work. Since python is required for our integration tests anyway, I could probably go ahead and have the configure_dev script guarantee the correct version of python is installed for mule. Then it'll be a one liner in the travis to say pip install mulecli